### PR TITLE
[tune] API revamp fix

### PR DESCRIPTION
**Pull request by richardliaw**
## Why are these changes needed?

Deprecates a couple config values to reduce the number of args in tune.run.

TODO:
 - [x] docs
 - [x] Fix test_sync

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: python/ray/tune/tests/test_sync.py [hunk 1]
```diff
@@ -167,18 +167,20 @@ def trainable(config):
         def counter(local, remote):
             mock()
 
+        sync_config = tune.SyncConfig(upload_dir="test", sync_to_cloud=counter)
         tune.syncer.CLOUD_SYNC_PERIOD = 1
+        os.environ["TUNE_GLOBAL_CHECKPOINT_S"] = "0"
```
Comment (amogkam) [inline python/ray/tune/tests/test_sync.py:172]: should this be 0.5 or does it not matter?

File: python/ray/tune/tune.py [hunk 2]
```diff
@@ -230,45 +257,23 @@ def run(run_or_experiment,
...
-            LOCAL/True restores the checkpoint from the local_checkpoint_dir.
-            REMOTE restores the checkpoint from remote_checkpoint_dir.
-            PROMPT provides CLI feedback. False forces a new
-            experiment. If resume is set but checkpoint does not exist,
+        resume (str|bool): One of "LOCAL", "REMOTE", "PROMPT", "ERRORED_ONLY",
+            or bool. LOCAL/True restores the checkpoint from the
+            local_checkpoint_dir, determined
+            by `name` and `local_dir`. REMOTE restores the checkpoint
+            from remote_checkpoint_dir. PROMPT provides CLI feedback.
+            False forces a new experiment. ERRORED_ONLY resets and reruns
```
Comment (amogkam) [inline python/ray/tune/tune.py:266]: Does ERRORED_ONLY use local checkpoint directory by default? Is there no need to support both ERRORED_ONLY+LOCAL and ERRORED_ONLY+REMOTE?

File: python/ray/tune/utils/util.py [hunk 3]
```diff
@@ -152,6 +153,17 @@ def flush(self, *args, **kwargs):
         self.stream2.flush(*args, **kwargs)
 
 
+def env_integer(key, default):
```
Comment (amogkam) [inline python/ray/tune/utils/util.py:156]: is there any reason not move this into ray_constants in this PR?

**Review by amogkam — APPROVED** [review]
LGTM- just left a few questions.

File: python/ray/tune/tests/test_sync.py (hunk 1 above)
Comment (richardliaw) [reply python/ray/tune/tests/test_sync.py:172]: it should be ok as tests pass for me but i’ll leave a comment.

File: python/ray/tune/tune.py (hunk 2 above)
Comment (richardliaw) [reply python/ray/tune/tune.py:266]: I think only supporting local is ok for now; remote is just added sugar/nice to have - I’ll push a comment.

File: python/ray/tune/utils/util.py (hunk 3 above)
Comment (richardliaw) [reply python/ray/tune/utils/util.py:156]: Good question - I wanted to get this PR in without needing to wait for Ray CI + and didnt want to risk breaking ray core.

File: python/ray/tune/cluster_info.py [hunk 4]
```diff
@@ -2,6 +2,10 @@
 import os
 
 
+def is_ray_cluster():
+    return os.path.exists(os.path.expanduser("~/ray_bootstrap_config.yaml"))
+
```
Comment (krfricke) [inline python/ray/tune/cluster_info.py:7]: Is this always true for ray clusters?

File: python/ray/tune/trial_runner.py [hunk 5]
```diff
@@ -168,7 +163,7 @@ def __init__(self,
 
         self._server = None
         self._server_port = server_port
-        if launch_web_server:
+        if server_port is not None:
```
Comment (krfricke) [inline python/ray/tune/trial_runner.py:166]: remove `launch_web_server` argument from init?

File: python/ray/tune/tune.py [hunk 6]
```diff
@@ -166,14 +201,24 @@ def run(run_or_experiment,
...
             Defaults to ``~/ray_results``.
-        upload_dir (str): Optional URI to sync training results and checkpoints
-            to (e.g. ``s3://bucket`` or ``gs://bucket``).
-        trial_name_creator (Callable[[Trial], str]): Optional function
-            for generating the trial string representation.
-        trial_dirname_creator (Callable[[Trial], str]): Function
-            for generating the trial dirname. This function should take
-            in a Trial object and return a string representing the
-            name of the directory. The return value cannot be a path.
+        search_alg (Searcher): Search algorithm for optimization.
```
Comment (krfricke) [inline python/ray/tune/tune.py:204]: The order of the arguments in the docstring should match the order of the arguments in the init

File: python/ray/tune/tune.py [hunk 7]
```diff
@@ -277,12 +281,35 @@ def run(run_or_experiment,
     Raises:
         TuneError: Any trials failed and `raise_on_failed_trial` is True.
     """
+    if global_checkpoint_period:
+        raise ValueError("global_checkpoint_period is deprecated. "
+                         "Set 'TUNE_GLOBAL_CHECKPOINT_S' instead.")
+    if queue_trials:
+        raise ValueError(
+            "queue_trials is deprecated. "
+            "Set 'TUNE_DISABLE_QUEUE_TRIALS=1' instead to disable queuing"
```
Comment (krfricke) [inline python/ray/tune/tune.py:290]: Might be a nit, but it would be good to explicitly say tht TUNE_DISABLE_QUEUE_TRIALS is an environment variable

**Review by krfricke — COMMENTED** [review]
Generally looks good!

File: python/ray/tune/cluster_info.py (hunk 4 above)
Comment (richardliaw) [reply python/ray/tune/cluster_info.py:7]: yep - for all clusters started with the cluster launcher, left a comment

File: python/ray/tune/trial_runner.py (hunk 5 above)
Comment (richardliaw) [reply python/ray/tune/trial_runner.py:164]: done

File: python/ray/tune/tune.py (hunk 6 above)
Comment (richardliaw) [reply python/ray/tune/tune.py:171]: done.

File: python/ray/tune/tests/test_tune_server.py [hunk 8]
```diff
@@ -29,7 +29,7 @@ class TuneServerSuite(unittest.TestCase):
     def basicSetup(self):
         ray.init(num_cpus=4, num_gpus=1)
         port = get_valid_port()
-        self.runner = TrialRunner(launch_web_server=True, server_port=port)
+        self.runner = TrialRunner(erver_port=port)
```
Comment (richardliaw) [inline python/ray/tune/tests/test_tune_server.py:32]:
```suggestion
        self.runner = TrialRunner(server_port=port)
```

**Comment by ericl** [comment]
Can we revert this?

I don't think removing queue trials as an option is tenable right now until we have placement groups fully supported. This will cause weird behavior for a lot of workloads (including all of RLlib). Also, now RLlib prints this weird warning message "TUNE_DISABLE_QUEUE_TRIALS detected"

**Pull request merged**